### PR TITLE
Tornado Cash: Fix sources, lowercase and improve materialization strategy

### PR DESCRIPTION
**Pull request by soispoke**
Brief comments on the purpose of your changes:

*For Dune Engine V2*
I've checked that:

* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

When you are ready for a review, tag duneanalytics/data-experience. We will re-open your forked pull request as an internal pull request. Then your spells will run in dbt and the logs will be avaiable in Github Actions DBT Slim CI. This job will only run the models and tests changed by your PR compared to the production project.

### CODE REVIEW

File: models/tornado_cash/tornado_cash_deposits.sql [hunk 1]
```diff
@@ -35,13 +35,13 @@ FROM
                 AND et.block_time >= (select min(evt_block_time) from {{ source('tornado_cash_ethereum','eth_evt_Deposit') }})
                 {% endif %}
                 {% if is_incremental() %}
-                AND et.block_time >= (select max(block_time) from {{ this }})
+                AND et.block_time >= date_trunc("day", now() - interval '1 week')
```
Comment (aalan3) [inline models/tornado_cash/tornado_cash_deposits.sql:38]: Shouldn't the `LEFT JOIN` above be an `INNER JOIN` instead? And then you could drop the where filter below.

File: models/tornado_cash/tornado_cash_deposits.sql [hunk 2]
```diff
@@ -35,13 +35,13 @@ FROM
                 AND et.block_time >= (select min(evt_block_time) from {{ source('tornado_cash_ethereum','eth_evt_Deposit') }})
```
Comment (aalan3) [inline models/tornado_cash/tornado_cash_deposits.sql:35]: Can't comment on the actual line but line 32  `source('ethereum','transactions_0006'`) should be ` source('ethereum','transactions')`.

Comment (jeff-dude) [reply models/tornado_cash/tornado_cash_deposits.sql:35]: i believe that source was used specifically for performance reasons, but agreed to revert back if we're comfortable with new performance enhancement changes (could then also remove from source file)

File: models/tornado_cash/tornado_cash_deposits.sql (hunk 1 above)
Comment (jeff-dude) [reply models/tornado_cash/tornado_cash_deposits.sql:38]: added this common check to review doc, but many existing queries from v1 will have left joins to look out for

Comment (aalan3) [reply models/tornado_cash/tornado_cash_deposits.sql:38]: There are cases where you'd want to do a left join as I discovered [here](https://github.com/duneanalytics/spellbook/pull/1465/files#diff-5bf74bdf14e57e88e2a0385fed61e5f0a5ba0d554f05f56bf255e413ca9eec7aR86). But generally it seems like it should be inner.

**Comment by jeff-dude** [comment]
lgtm after changes applied from alan feedback

**Comment by hildobby** [comment]
Discussed this in DMs with @jeff-dude, imo naming should be consistent across the board and since `avalanche_c` is avalanche's default blockchain name, all abstractions should also stick with it for consistency, don't you guys think?

PS: Apologies for not changing the names myself when doing the first tornado cash PR, I will make sure to be careful on names being lowercase going forward!

**Review by soispoke — COMMENTED** [review]
mades some small changes regarding lowercasing blockchains and moving from avalanche to avalanche_c for consistency

File: macros/alter_table_properties.sql [hunk 3]
```diff
@@ -560,17 +560,17 @@ ALTER TABLE airdrop_optimism.addresses SET TBLPROPERTIES ('dune.public'='true',
 {% endset %}
 
 {% set tornado_cash_deposits %}
-ALTER TABLE tornado_cash.deposits SET TBLPROPERTIES ('dune.public'='true',
-                                                'dune.data_explorer.blockchains'='["ethereum", "bnb", "avalanche", "gnosis", "optimism", "arbitrum"]',
+ALTER VIEW tornado_cash.deposits SET TBLPROPERTIES ('dune.public'='true',
```
Comment (jeff-dude) [inline macros/alter_table_properties.sql:563]: shouldn't these both be 'table'?

File: models/tornado_cash/tornado_cash_deposits.sql (hunk 2 above)
Comment (jeff-dude) [reply models/tornado_cash/tornado_cash_deposits.sql:35]: @soispoke can you update this?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by jeff-dude** [comment]
to @aalan3 point, should we use this PR to update all joins to base tables as inner join?

File: models/tornado_cash/tornado_cash_deposits.sql (hunk 2 above)
Comment (soispoke) [reply models/tornado_cash/tornado_cash_deposits.sql:35]: @jeff-dude Done

File: macros/alter_table_properties.sql (hunk 3 above)
Comment (soispoke) [reply macros/alter_table_properties.sql:581]: You're right they should, pushed a fix

**Review by soispoke — COMMENTED** [review]
@jeff-dude Done, updated all joins with base tables

**Review by aalan3 — APPROVED** [review]

**Comment by soispoke** [comment]
Thanks !

File: macros/alter_table_properties.sql [hunk 4]
```diff
@@ -579,16 +579,16 @@ ALTER TABLE airdrop_optimism.addresses SET TBLPROPERTIES ('dune.public'='true',
...
                                                 'dune.data_explorer.category'='abstraction',
                                                 'dune.data_explorer.abstraction.type'='project',
                                                 'dune.data_explorer.abstraction.name'='tornado_cash',
                                                 'dune.data_explorer.contributors'='["hildobby", "dot2dotseurat"]');
 {% endset %}
 
 {% set tornado_cash_withdrawals %}
-ALTER TABLE tornado_cash.withdrawals SET TBLPROPERTIES ('dune.public'='true',
-                                                'dune.data_explorer.blockchains'='["ethereum", "bnb", "avalanche", "gnosis", "optimism", "arbitrum"]',
+ALTER VIEW tornado_cash.withdrawals SET TBLPROPERTIES ('dune.public'='true',
```
Comment (jeff-dude) [inline macros/alter_table_properties.sql:590]: @soispoke looks like this one didn't get updated back to 'table'

Comment (aalan3) [reply macros/alter_table_properties.sql:590]: Was fixed here https://github.com/duneanalytics/spellbook/pull/1475/files#diff-89d8392c4db5b996c368bef69b027f7ccd442b9135924630007065e3a1d5569cR590

**Pull request merged**